hbase-server/src/main/java/org/apache/hadoop/hbase/replication/regionserver/ReplicationSourceManager.java

```diff
@@ -579,8 +579,8 @@ private void interruptOrAbortWhenFail(ReplicationQueueOperation op) {
       if (e.getCause() != null && e.getCause() instanceof KeeperException.SystemErrorException
           && e.getCause().getCause() != null && e.getCause()
           .getCause() instanceof InterruptedException) {
-        throw new RuntimeException(
-            "Thread is interrupted, the replication source may be terminated");
+        LOG.info("Thread is interrupted, the replication source may be terminated");
+        return;
```

Comment:
You sure you don't want to propagate the `InterruptedException`? Looks like `run` loop in `ReplicationSourceShipper` is designed to handle that exception. `return` here would continue the thread executing, which I think is not what we want.

Reply:
the interrupt flap has been set up in the zkwatcher layer and upper layer handles this flag nicely.

      zkw.interruptedException(ie);
https://github.com/apache/hbase/blob/master/hbase-zookeeper/src/main/java/org/apache/hadoop/hbase/zookeeper/ZKWatcher.java#L637, in the outside loop, it will handle this interrupt flag and terminate the thread.

Inside  isActive() of the following loop, it will return false if isInterrupted flag is true.

https://github.com/apache/hbase/blob/master/hbase-server/src/main/java/org/apache/hadoop/hbase/replication/regionserver/ReplicationSourceShipper.java#L101

Reply:
I have manually tested the code path by injecting an InterruptException at zkwatcher and made sure thread exits and  no region server abort.

Reply:
Thanks for checking.